xbmc/ServiceBroker.cpp

```diff
@@ -27,6 +27,21 @@ CServiceBroker::~CServiceBroker()
 {
 }
 
+std::shared_ptr<CAppParams> CServiceBroker::GetAppParams()
+{
+  return g_serviceBroker.m_appParams;
```

Comment:
You don't seem to check for nullptr anywhere that `GetAppParams` is called. I believe it should always be available, however, do you think we should add a nullptr check here and throw or log if it is?

Reply:
App params is supposed to be available as the very first "service" and calling it when it is not yet available is a logical error in the application, so we should throw a logic_error. Will change this.